### PR TITLE
[Feature] Support cluster manage statements in new RBAC privilege framework

### LINKED ISSUE #13107: support show node info statements in new RBAC privilege framework
## Feature request

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by waittttting**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #13107

## Problem Summary(Required) ：

PrivilegeCheckerV2 support cluster manage statements
1. `show backends`, `show frontends`, `show broker` statements.
2. `show tablet`, `show transaction`.

Support upgrading the privilege of system.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 9 (88.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|6|7|85.71%|[220]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 14 (78.57%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|9|12|75.00%|[222, 231, 243]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 1]
```diff
@@ -195,5 +208,92 @@ public Void visitAlterResourceStatement(AlterResourceStmt statement, ConnectCont
             }
             return null;
         }
+
+        // ---------------------------------------- Show Node Info Statement---------------------------------------------
+        @Override
+        public Void visitShowBackendsStatement(ShowBackendsStmt statement, ConnectContext context) {
+            return checkShowNodePrivilege(context);
```
Comment (padmejin) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:215]: Should check for either `node` or `operate`. Same with `show frontends`, `show brokers`.

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 2]
```diff
@@ -195,5 +208,92 @@ public Void visitAlterResourceStatement(AlterResourceStmt statement, ConnectCont
...
+        public Void visitAdminCheckTabletsStatement(AdminCheckTabletsStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        // ---------------------------------------- Common Privilege logic -------------------------------------------------
+        private Void checkStmtOperatePrivilege(ConnectContext context) {
+            if (!PrivilegeManager.checkSystemAction(context, PrivilegeType.SystemAction.OPERATE)) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "OPERATE");
+            }
+            return null;
```
Comment (padmejin) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:296]: what about the following statements?
* kill <id>
* alter system xx
* cancel alter system
* show compute nodes
* show proc
* show variables like
* set

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 28 / 31 (90.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|17|20|85.00%|[230, 239, 294]|
|:large_blue_circle:|com/starrocks/mysql/privilege/AuthUpgrader.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 19 / 22 (86.36%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|17|20|85.00%|[230, 239, 294]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java (hunk 1 above)
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:256]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java (hunk 2 above)
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:566]: done

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java [hunk 3]
```diff
@@ -491,4 +509,139 @@ public void testDatabaseStmt() throws Exception {
...
+                "show broker",
+                "grant OPERATE on system to test",
+                "revoke OPERATE on system from test",
+                "Access denied; you need (at least one of) the OPERATE/NODE privilege(s) for this operation");
+    }
+
+    @Test
+    public void testShowTransactionAndTabletStmt() throws Exception {
+
+        verifyGrantRevoke(
```
Comment (nshangyiming) [inline fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java:538]: where is show txn?

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java [hunk 4]
```diff
@@ -104,6 +104,24 @@ private static void verifyGrantRevoke(String sql, String grantSql, String revoke
...
+    private static void verifyNODEAndGRANT(String sql, String expectError) throws Exception {
+        ctxToRoot();
+        ConnectContext ctx = starRocksAssert.getCtx();
+        StatementBase statement = UtFrameUtils.parseStmtWithNewParser(sql, ctx);
+        // user 'root' has GRANT/NODE privilege
+        PrivilegeCheckerV2.check(statement, starRocksAssert.getCtx());
+
+        try {
+            ctxToTestUser();
+            // user 'test' not has GRANT/NOD privilege
```
Comment (nshangyiming) [inline fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java:116]: typo: NODE

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java [hunk 5]
```diff
@@ -491,4 +509,139 @@ public void testDatabaseStmt() throws Exception {
                 "revoke ALTER on database " + testDbName + " from test",
                 "Access denied for user 'test' to database '" + testDbName + "'");
     }
+    
+    @Test
+    public void testShowNodeStmt() throws Exception {
+
+        verifyGrantRevoke(
```
Comment (nshangyiming) [inline fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java:516]: these three show statements also need to test OPERATE privilege.

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java [hunk 6]
```diff
@@ -491,4 +509,139 @@ public void testDatabaseStmt() throws Exception {
...
+                "show tablet from example_db.example_table",
+                "grant OPERATE on system to test",
+                "revoke OPERATE on system from test",
+                "Access denied; you need (at least one of) the OPERATE privilege(s) for this operation");
+    }
+
+    @Test
+    public void testAdminOperateStmt() throws Exception {
```
Comment (nshangyiming) [inline fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java:546]: no test case for  show compute nodes

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 7]
```diff
@@ -291,6 +317,35 @@ public Void visitInstallPluginStatement(InstallPluginStmt statement, ConnectCont
...
+        public Void visitShowBrokerStatement(ShowBrokerStmt statement, ConnectContext context) {
+            return checkShowNodePrivilege(context);
+        }
+
+        @Override
+        public Void visitShowComputeNodes(ShowComputeNodesStmt statement, ConnectContext context) {
+            return checkShowNodePrivilege(context);
+        }
+
+        private Void checkShowNodePrivilege(ConnectContext context) {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:341]: plz put the check method with other check methods in this file ,like `checkDbAction` etc.

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java [hunk 8]
```diff
@@ -514,5 +576,121 @@ public Void visitDropTableStatement(DropTableStmt statement, ConnectContext sess
...
+                }
+                if (setVer.getType().equals(SetType.GLOBAL)) {
+                    checkStmtOperatePrivilege(context);
+                }
+            });
+            return null;
+        }
+
+        // ---------------------------------------- Common Privilege logic -------------------------------------------------
+        private Void checkStmtOperatePrivilege(ConnectContext context) {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:682]: plz put the check method with other check methods in this file ,like `checkDbAction` etc.

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java (hunk 4 above)
Comment (waittttting) [reply fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java:116]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java (hunk 7 above)
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:341]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java (hunk 8 above)
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:682]: done

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java (hunk 6 above)
Comment (waittttting) [reply fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java:546]: done

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java (hunk 5 above)
Comment (waittttting) [reply fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java:516]: done

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java (hunk 3 above)
Comment (waittttting) [reply fe/fe-core/src/test/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2Test.java:561]: done

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 55 / 70 (78.57%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|4|00.00%|[670, 673, 674, 676]|
|:large_blue_circle:|com/starrocks/catalog/ResourceMgr.java|4|5|80.00%|[308]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|49|59|83.05%|[155, 161, 356, 596, 689, 690, 694, 699, 700, 702]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 60 / 77 (77.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|4|00.00%|[670, 673, 674, 676]|
|:large_blue_circle:|com/starrocks/sql/ast/SetPassVar.java|2|3|66.67%|[76]|
|:large_blue_circle:|com/starrocks/sql/ast/SetVar.java|3|4|75.00%|[98]|
|:large_blue_circle:|com/starrocks/catalog/ResourceMgr.java|4|5|80.00%|[308]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|49|59|83.05%|[155, 161, 356, 596, 689, 690, 694, 699, 700, 702]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|2|2|100.00%|[]|

**Review by nshangyiming — DISMISSED** [review]

**Review by gengjun-git — DISMISSED** [review]

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by wangruin** [comment]
run starrocks_admit_test

**Comment by wangruin** [comment]
run starrocks_admit_test

**Comment by dulong41** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 60 / 77 (77.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|4|00.00%|[670, 673, 674, 676]|
|:large_blue_circle:|com/starrocks/sql/ast/SetPassVar.java|2|3|66.67%|[76]|
|:large_blue_circle:|com/starrocks/sql/ast/SetVar.java|3|4|75.00%|[98]|
|:large_blue_circle:|com/starrocks/catalog/ResourceMgr.java|4|5|80.00%|[308]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|49|59|83.05%|[155, 161, 356, 596, 689, 690, 694, 699, 700, 702]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|2|2|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**